cmd_clean/main.go

```diff
@@ -33,6 +36,22 @@ var (
 	cluster     string
 )
 
+type Transport struct {
+	Base http.RoundTripper
+}
+
+func (t *Transport) RoundTrip(r *http.Request) (*http.Response, error) {
+	log.V(1).Info("http request", "method", r.Method, "body", r.Body, "url", r.URL)
+	r.SetBasicAuth(nsxUser, nsxPasswd)
+	return t.base().RoundTrip(r)
+}
+func (t *Transport) base() http.RoundTripper {
+	if t.Base != nil {
+		return t.Base
+	}
+	return http.DefaultTransport
+}
+
 func main() {
 	flag.StringVar(&vcEndpoint, "vc-endpoint", "", "nsx manager ip")
 	flag.StringVar(&vcSsoDomain, "vc-sso-domain", "", "nsx manager ip")
```

Comment:
Better to add a flag use-external-http and set default false, if false, use previous logic, else use httpclient, thus not affecting the existed logic.